lib/utils/normalizeSelector.mjs

```diff
@@ -0,0 +1,84 @@
+import parser from 'postcss-selector-parser';
+
+/** @import { Container, Node, Root, Selector } from 'postcss-selector-parser' */
+
+/**
+ * Normalize a selector so that it can be more easily compared to other selectors.
+ *
+ * @template {Root | Selector} T
+ * @param {T} node
+ * @returns {T}
+ */
+export default function normalizeSelector(node) {
+	normalizeSelectorNodeWhitespaceAndComments(node);
+
+	node.walk(normalizeSelectorNodeWhitespaceAndComments);
+
+	normalizeSelectorNodesOrder(node);
+
+	return node;
+}
+
+/**
+ * @param {Node} node
+ */
+function normalizeSelectorNodeWhitespaceAndComments(node) {
+	if (parser.isComment(node)) {
+		node.remove();
+
+		return;
+	}
+
+	node.rawSpaceBefore = '';
+	node.rawSpaceAfter = '';
+
+	node.spaces.before = '';
+	node.spaces.after = '';
+
+	if (parser.isAttribute(node)) {
+		delete node.spaces.insensitive;
+		delete node.spaces.operator;
+		delete node.spaces.attribute;
+		delete node.spaces.value;
+
+		delete node.raws.spaces;
+	}
```

Comment:
This was a lot less code and complexity than I was expecting it to be 😅 

The typings in `postcss-selector-parser` are really complex and gave me the false impression that each node has some specific fields, while in reality it limited to attributes 🎉